Handle punctuation

jhchen opened the pull request:
Following up on #5 with something to take a look at. If the approach is acceptable perhaps it would be better to move to snake_case.ex instead since it seems like the other cases depend on that one (and update the other tests).

I tried running `mix credo --strict` and `mix dialyzer` but they errorred on the main sobolevn/recase master branch as well so I don't think this PR introduces any new cod equality issues. If you're not seeing the same I can provide more details on what I'm seeing.

Last thing (and perhaps this should be separate) but using Lodash's implementation as a guide, they also split up strings with digits ex you can try `_.snakeCase("ab-cD-e0-Df-GH-I1-2j-3K-45") === "ab_c_d_e_0_df_gh_i_1_2_j_3_k_45"` in the developer console in lodash.com.

sobolevn commented:
@jhchen Thanks for your work! I appreciate it.

What about `snake_case`? Is it any different from `kebab`?
If it should behave in the same manner - it means, that we should fix it as well (or fix just `snake_case` since many other cases rely on it).

I think we should discuss `lodash` way of handling `kebab` case as a separate issue. Will you, please, open it?

jhchen commented:
Okay I updated the code and PR to implement the punctuation fix in snake case instead, updated the affected tests and opened a new issue for the digits question.

sobolevn commented:
@jhchen you are awesome!

jhchen commented:
Thanks! You have any thoughts on when to release a new version?

sobolevn commented:
I am actually releasing it in a few hours. 
Right now I am trying to fix some minor issues with the CI and stuff.

sobolevn commented:
@jhchen done.
